qiskit_nature/second_q/operators/polynomial_tensor.py

```diff
@@ -0,0 +1,208 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""PolynomialTensor class"""
+
+from __future__ import annotations
+from typing import Dict, Iterator
+from collections.abc import Mapping
+from numbers import Number
+import numpy as np
+from qiskit.quantum_info.operators.mixins import (
+    LinearMixin,
+    AdjointMixin,
+    TolerancesMixin,
+)
+
+
+class PolynomialTensor(LinearMixin, AdjointMixin, TolerancesMixin, Mapping):
+    """PolynomialTensor class"""
```

Comment:
For a future reviewer: I will flesh out this docstring in an upcoming PR when working on #666 :+1: